hikari/impl/shard.py

```diff
@@ -184,21 +196,16 @@ async def _receive_and_check(self, timeout: typing.Optional[float], /) -> str:
             return message.data
 
         if message.type == aiohttp.WSMsgType.BINARY:
+            if message.data.endswith(_ZLIB_SUFFIX):
+                return self.zlib.decompress(message.data).decode("utf-8")
```

Comment:
Since we already have the while loop in the `_receive_and_check_complete_zlib_package`, wouldnt this just be duplicated code?

*If anything, `_receive_and_check_complete_zlib_package` could just be merged into here*